### ISSUE TITLE
interrupted speech is not continued

### ISSUE THREAD

**Issue by telic** [emmental/isetools#11]
The XMLWriter is not properly resuming speeches on a line after they have been interrupted by stage directions.

For example, this input IML document:
```sgml
<work>
<scene n="1">
<S>speech <SD t="exit">interject</SD> continues</S>
</scene>
</work>
```
is incorrectly output as:
```xml
<work xmlns="http://internetshakespeare.uvic.ca/exist/rest/db/apps/iseapp/content/schema/text/documentClass/dramaticWork.rng">
<scene n="1"/>
<l><s k="1">speech </s><sd t="exit">interject</sd> continues</l>
</work>
```
where the `s` is missing from the "continues" part.

This bug also occurs when the speech starts on an earlier line, but does not affect speech parts after the line with the stage direction.

Notably, interrupting a speech with `MARG` is handled correctly.

**Comment by telic** [comment]
This may be a different issue all together, but speeches are also not being resumed when crossed by an `<add>`, eg:

```sgml
<work>
<scene n="1">
<L/><add><S>test</add> continue</S>
</scene>
</work>
```

The word "continue" should be included in the speech, but currently is not. Swapping the source order of the opening `add` and `S` tags results in the expected output rendering.

**Comment by emmental** [comment]
done: aa9534c

**Comment by emmental** [comment]
Sorry, I fixed the ADD issue too: e6d0935
